colour/hints/__init__.py

```diff
@@ -149,18 +149,18 @@ def x(self) -> NDArray:  # noqa: D102
         ...
 
     @x.setter
-    def x(self, value: ArrayLike) -> None: ...
+    def x(self, value: ArrayLike, /) -> None: ...
 
     @property
     def y(self) -> NDArray:  # noqa: D102
         ...
 
     @y.setter
-    def y(self, value: ArrayLike) -> None: ...
+    def y(self, value: ArrayLike, /) -> None: ...
 
     def __init__(self, *args: Any, **kwargs: Any) -> None: ...  # pragma: no cover
 
-    def __call__(self, x: ArrayLike) -> NDArray:  # noqa: D102
+    def __call__(self, x: NDArrayFloat) -> NDArray:  # noqa: D102
```

Comment:
input positions are contravariant, so assigning to this protocol, requires that the parameter types of that type are *less* strict than that of this `Protocol`. Put differently; narrowing the `ArrayLike` parameter type to `NDArrayFloat` results in a broader `Protocol` type.